bpf/bpf_lxc.c

```diff
@@ -110,10 +140,18 @@ static __always_inline int __per_packet_lb_svc_xlate_4(void *ctx, struct iphdr *
 		 * redirect services based on user configured policies. Per packet LB should
 		 * not override LB decisions made for local-redirect services in bpf_sock.
 		 */
-#if defined(ENABLE_LOCAL_REDIRECT_POLICY) && defined(ENABLE_SOCKET_LB_FULL)
+#if defined(ENABLE_LOCAL_REDIRECT_POLICY)
+	#if defined(ENABLE_SOCKET_LB_FULL)
 		if (unlikely(lb4_svc_is_localredirect(svc)))
 			goto skip_service_lookup;
-#endif /* ENABLE_LOCAL_REDIRECT_POLICY && ENABLE_SOCKET_LB_FULL */
+	#endif /* ENABLE_SOCKET_LB_FULL */
+	#ifdef HAVE_NETNS_COOKIE
+		if (unlikely(lb4_svc_is_localredirect(svc)) &&
+		    lb4_skip_xlate_from_ctx_to_svc(ENDPOINT_NETNS_COOKIE, tuple.daddr, tuple.sport))
+			goto skip_service_lookup;
+	#endif
```

Comment:
Can we move this logic to the code where the datapath detects [loopback traffic](https://github.com/cilium/cilium/blob/main/bpf/lib/lb.h#L1638)? We can avoid the penalty of executing this for regular traffic on a per packet basis.

Reply:
Made the suggested change. But lb4/6_local() is invoked from bpf_host.c and bpf_overlay.c as well. I pass an invalid value for netns cookie for those invocations.